shm/Map.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include "Island.hpp"
+
+constexpr size_t defaultXsize = 10;
+constexpr size_t defaultYsize = 11;
+constexpr int defaultIslandsAmount = 10;
```

Comment:
Can "amount" be <0 ? I mean - why it's an `int`?

Reply:
+1